tarodb/sqlc/queries/assets.sql

```diff
@@ -212,22 +212,27 @@ SELECT
 FROM assets
 JOIN genesis_info_view
     ON assets.genesis_id = genesis_info_view.gen_asset_id AND
-        (length(hex(sqlc.narg('asset_id_filter'))) = 0 OR genesis_info_view.asset_id = sqlc.narg('asset_id_filter'))
+      (genesis_info_view.asset_id = sqlc.narg('asset_id_filter') OR 
```

Comment:
Oh wow, you actually got the intended version working?

IIRC, I tried something like this in the past, but was never able to get the default case to return true.

Reply:
Yeah, at least it looks like it's working. Do you remember what error you got? Was it at compile time? I think it's just the order of the `sqlc.narg`, it first needs to be compared against a field so sqlc can estimate its type. If you first use it against `IS NULL`, it cannot extract the type.